src/screens/Settings/ChangePassword.js

```diff
@@ -37,21 +41,34 @@ const initialValues = {
 export default class ChangePassword extends Component {
   static navigationOptions = () => ({ title: "Change Password" });
 
-  state = { ...initialValues };
-
-  onSubmit = async (values) => {
+  onSubmit = async (values, { resetForm }) => {
     let response;
+    let authResponse;
     try {
       if (values.newPassword1 !== values.newPassword2) {
         throw new Error("New passwords don't match.");
       }
-
       const user = await checkIfLoggedIn();
       if (!user) throw new Error("Not logged in.");
 
-      if (true) throw new Error("Incorrect password.");
+      // Attempt to login with current password
+      authResponse = await authServices.login({
+        username: user.credentials.email,
+        password: values.password
+      });
+      if (authResponse instanceof Error) throw new Error("Could not authorize current credentials.");
 
-      this.props.nav.navigate("SettingsPage", {
+      let data = {
+        _id: authResponse.credentials.id,
+        password: values.newPassword1,
+        user: authResponse
+      }
+      response = await userServices.put(data);
+      if (response instanceof Error) throw response;
+
+      asyncStore.save("user", JSON.stringify(response));
+      this.clearForm(resetForm);
```

Comment:
I don't think you need to extract this as a function.

Reply:
clearForm? It was declared as a function in the original ReportForm, which I then turned into the EventForm component, so I copied the pattern that was there. I can check but I assumed it was necessary...